src/test/unit/common/languageserver/languageserver.test.ts

```diff
@@ -0,0 +1,153 @@
+import assert, { deepStrictEqual } from 'assert';
+import { LanguageServer } from '../../../../snyk/common/languageserver/languageserver';
+import { ILanguageClientAdapter } from '../../../../snyk/common/vscode/languageClient';
+import { IVSCodeWorkspace } from '../../../../snyk/common/vscode/workspace';
+import { stubWorkspaceConfiguration } from '../../mocks/workspace.mock';
+
+suite('languageserver', () => {
+  let extensionContext: ExtensionContext;
+  let configuration: IConfiguration;
+  let languageServer: LanguageServer;
+  setup(() => {
+    extensionContext = {
+      secrets: {
+        store: (_key: string, _value: string) => Promise.resolve(),
+        get: () => Promise.resolve(),
+        delete: () => Promise.resolve(),
+      },
+    } as unknown as ExtensionContext;
+
+    SecretStorageAdapter.init(extensionContext);
+    configuration = {
+      getCustomCliPath(): string | undefined {
+        return 'testPath';
+      },
+      getToken(): Promise<string | undefined> {
+        return Promise.resolve('testToken');
+      },
+      shouldReportEvents: true,
+      shouldReportErrors: true,
+      getPreviewFeatures(): PreviewFeatures {
+        return {
+          advisor: false,
+          reportFalsePositives: false,
+          lsAuthenticate: true,
+        };
+      },
+      getSnykLanguageServerPath(): string {
+        return 'testPath';
+      },
+    } as unknown as IConfiguration;
+  });
+
+  teardown(() => {
+    sinon.restore();
+  });
+
+  test('LanguageServer should provide correct initialization options', async () => {
+    languageServer = new LanguageServer(
+      extensionContext,
+      configuration,
+      {} as unknown as ILanguageClientAdapter,
+      {} as unknown as IVSCodeWorkspace,
+    );
+    const expectedInitializationOptions = {
+      activateSnykCode: 'false',
+      activateSnykOpenSource: 'false',
+      activateSnykIac: 'false',
+      token: 'testToken',
+      cliPath: 'testPath',
+      enableTelemetry: 'true',
+      sendErrorReports: 'true',
+      integrationName: 'VS_CODE',
+      integrationVersion: '0.0.0',
+    };
+
+    deepStrictEqual(await languageServer.getInitializationOptions(), expectedInitializationOptions);
+  });
+
+  test('LanguageServer starts language client if preview enabled', async () => {
+    const lc = sinon.spy({
+      start(): Promise<void> {
+        return Promise.resolve();
+      },
+    } as unknown as LanguageClient);
+
+    const lca = sinon.spy({
+      create(): LanguageClient {
+        return lc as unknown as LanguageClient;
+      },
+    });
+
+    languageServer = new LanguageServer(
+      extensionContext,
+      configuration,
+      lca as unknown as ILanguageClientAdapter,
+      stubWorkspaceConfiguration('http.proxy', undefined),
+    );
+    await languageServer.start();
+    sinon.assert.called(lca.create);
+    sinon.assert.called(lc.start);
+    sinon.verify();
```

Comment:
I think you don't need `verify` because you don't set any expectation on mock. Instead, you're asserting in the two lines above. May be wrong here, please double check.